pkg/resources/row_access_policy.go

```diff
@@ -117,104 +69,84 @@ func RowAccessPolicy() *schema.Resource {
 // CreateRowAccessPolicy implements schema.CreateFunc.
 func CreateRowAccessPolicy(d *schema.ResourceData, meta interface{}) error {
 	db := meta.(*sql.DB)
+	ctx := context.Background()
+	client := sdk.NewClientFromDB(db)
+
+	databaseName := d.Get("database").(string)
+	schemaName := d.Get("schema").(string)
 	name := d.Get("name").(string)
-	database := d.Get("database").(string)
-	schema := d.Get("schema").(string)
-	signature := d.Get("signature").(map[string]interface{})
+	id := sdk.NewSchemaObjectIdentifier(databaseName, schemaName, name)
+
+	signature := d.Get("signature").(map[string]any)
 	rowAccessExpression := d.Get("row_access_expression").(string)
 
-	builder := snowflake.RowAccessPolicy(name, database, schema)
+	args := make([]sdk.CreateRowAccessPolicyArgsRequest, 0)
+	for k, v := range signature {
+		dataType, err := sdk.ToDataType(v.(string))
```

Comment:
It will. That's why there is TODO with SNOW-1020074 left.

I will for now remove from here ToDataType part (so that it's not stripped) but still, describe strips this info so we will still end up with infinite plan without the diff suppression (which SNOW-1020074 is about).